File: tiomacacomanager/src/model/SongRequest.ts
export interface SongRequest{
    request_time: string,
    sender: string,
    songname: string,
    vid_url: string,
    song_thumbnail: string
    requestid?: string,
}

export function getDefaultSongRequest(): SongRequest {
    return {
    request_time: "",
    sender: "",
    songname: "",
    vid_url: "",
    song_thumbnail:""
  }
}

File: tiomacacomanagerback/app/model/SongRequest.py
from typing import Optional
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, HttpUrl

class SongRequest(BaseModel):
    request_time: datetime
    sender: str
    songname: str
    vid_url: HttpUrl
    song_thumbnail: HttpUrl
    requestid: Optional[UUID] = None

    #jsonification params
    model_config = {
        "json_encoders": {
            HttpUrl: str,
            UUID: str,
            datetime: lambda dt: dt.isoformat()
        }
    }